csharp/src/Ice/ObjectAdapter.cs

```diff
@@ -166,25 +160,19 @@ public async ValueTask DisposeAsync()
 
             async Task PerformDisposeAsync()
             {
+                // Synchronously Dispose of the incoming connection factories to stop accepting new incoming requests
+                // or connections. This ensures that once DisposeAsync returns, no new requests will be dispatched.
+                Task[] tasks =
+                    _incomingConnectionFactories.Select(factory => factory.DisposeAsync().AsTask()).ToArray();
+
                 // Wait for activation to complete. This is necessary avoid out of order locator updates.
                 if (_activateTask != null)
                 {
                     await _activateTask.ConfigureAwait(false);
```

Comment:
Should we also ignore exceptions from _activateTask during dispose?

Reply:
Definitely, good catch! I'll fix this.